lib/base_service.js

```diff
@@ -31,13 +32,41 @@ class BaseService {
         this._logger = null;
         this._metrics = null;
         this._ratelimiter = null;
-        this.serviceReturns = null;
+        this._serviceReturns = null;
+        this._logUnhandledExceptionHandler = null;
+        this._unhandledRejectionHandler = null;
+
+        if (!this._logUnhandledExceptionHandler) {
+            this._logUnhandledExceptionHandler = (err) => {
+                this._logUnhandledException(err);
+                // Don't exit right away, allow logger to process message
+                setTimeout(() => {
+                    process.exit(1);
+                }, 100);
+            };
+        }
+        // Avoid recursion if there are bugs in the logging code.
+        this._inLogger = false;
+
+        if (!this._unhandledRejectionHandler) {
```

Comment:
Since it's a constructor this if will always evaluate to 'true'

Reply:
Removed